common/io/TimeoutManager.cpp

```diff
@@ -94,11 +94,7 @@ void TimeoutManager::CancelTimeout(timeout_id id) {
 
 TimeInterval TimeoutManager::ExecuteTimeouts(TimeStamp *now) {
   Event *e;
-  if (m_events.empty())
-    return TimeInterval();
-
```

Comment:
Code review wise, please could we put back the early return, yes it's technically redundant as the return after the while loop will catch it, but it's a coding style we use throughout so I think we'd prefer to leave it please.